OpenMage/magento-lts
# New feature: added partial postcode match to table rate shipping

### Description (*)
Added partial postcode match to table rate shipping. 

Needed to create shipping rules for Northern Ireland. Postcode should start with BT followed by anything else when the country is the UK. The only way possible in the old way was adding all possible exact postcodes. Now it is possible to add a wildcard asterix at the and of a partial postcode. For example BT* will match all postcodes starting with BT like BT1 1AA (a valid postcode from Northern Ireland). The same was already possible in the TAX rates zip/post code.

Related to issue: OpenMage/magento-lts#1486

### Related Pull Requests
OpenMage/magento-lts#1503


### Manual testing scenarios (*)
1. Add a partial postcode to the table rates. For example BT*.
2. All orders with postcodes starting with BT should now use this shipping rule.

### Questions or comments
The most specific postcode rule that is valid will be applied. So if you have the following postcodes for the same country and state:

- \*
- BT*
- BT1*
- BT1 1AA

And a customer places a order with "BT1 1BB" the line with "BT1*" will be applied. This is the most specific rule that matches because "BT1 1AA" doesn't match and "BT1*" is more specific than "BT*" and "*".

### Contribution checklist (*)
